www/js/app.js

```diff
@@ -1,22 +1,68 @@
 // Ionic E-Mission App
 
-// angular.module is a global place for creating, registering and retrieving Angular modules
-// 'emission' is the name of this angular module example (also set in a <body> attribute in index.html)
-// the 2nd parameter is an array of 'requires'
-// 'emission.services' is found in services.js
-// 'emission.controllers' is found in controllers.js
 'use strict';
 
-angular.module('emission', ['ionic',
+import angular from 'angular';
+import 'angular-animate';
+import 'angular-sanitize';
+import 'angular-translate';
+import '../manual_lib/angular-ui-router/angular-ui-router.js';
+import 'angular-local-storage';
+import 'angular-translate-loader-static-files';
+
+import 'moment';
+import 'moment-timezone';
+
+import i18next from 'i18next';
+
+import 'ionic-toast';
+import 'ionic-datepicker';
+import 'angular-simple-logger';
+
+import '../manual_lib/ionic/js/ionic.js';
+import '../manual_lib/ionic/js/ionic-angular.js';
+
+
+import en from '../i18n/en.json';
+import es from '../../locales/es/i18n/es.json';
+import fr from '../../locales/fr/i18n/fr.json';
+import it from '../../locales/it/i18n/it.json';
+const langs = { en, es, fr, it };
+
+let resources = {};
+for (const [lang, json] of Object.entries(langs)) {
+  resources[lang] = { translation: json }
+}
+
+const locales = !navigator?.length ? [navigator.language] : navigator.languages;
+let detectedLang;
+locales.forEach(locale => {
+  const lang = locale.trim().split(/-|_/)[0];
+  if (Object.keys(langs).includes(lang)) {
+    detectedLang = lang;
+  }
+});
+
+i18next.init({
+  debug: true,
+  resources,
```

Comment:
I don't think it matters. We're configuring the global instance of `i18next`, irrespective of AngularJS

I would; however, like to maybe put all this i18n initialization into another file rather than cluttering `app.js`

Reply:
TODO declutter i18next initialization

Reply:
I am OK with handling this later, but can we keep a  list of the TODOs somewhere where we don't forget them?